components/hierarchical-view/hierarchical-view-mixin.js

```diff
@@ -484,21 +477,23 @@ export const HierarchicalViewMixin = superclass => class extends superclass {
 	}
 
 	__onKeyDown(e) {
-		if (this.childView && e.keyCode === escapeKeyCode) {
+		if (this._childView && e.keyCode === escapeKeyCode) {
 			e.stopPropagation();
 			this.hide();
 			return;
 		}
 	}
 
 	__onShowStart(e) {
+		if (this.rootView) e.stopPropagation();
```

Comment:
The issue with the dropdowns trying to show the inner dropdown's menu seemed to be related to the show, hide, and resize events also propagating to the hierarchical-view that wrapped the inner menu. The `stopPropagation` seems to fix that, but if there are concerns that code could be relying on this behaviour (didn't see anything that seemed like it would cause an issue on a quick search) then we can also just do this if it `hasAttribute('root-view')`

Reply:
Seems reasonable.

Perhaps we should isolate all the hierarchical-view events at the root view. They were intended to be mostly internal to the hierarchy to manage the overall hierarchy state. The `d2l-menu` is interested in the `d2l-hierarchical-view-resize` event, but it's derived from `HierarchicalViewMixin` so it would continue to get the event on its host. It then dispatches it's own event, `d2l-menu-resize`, I think so that dropdowns can be resized.  

Looks like `d2l-filter` wires up for these events to call `stopPropagation` - [that code](https://github.com/BrightspaceUI/core/blob/main/components/filter/filter.js#L262) could probably be removed if we stopped all the events at the root view.